src/test/scala/hercules/api/services/MockBackend.scala

```diff
@@ -69,22 +62,36 @@ class MockBackend(
                     state
                 }
 
-                case m: RemoveFromFailedMessages =>
-                  state.manipulateState(m)
+                case m: RemoveFromFailedMessages => {
+                  state = state.manipulateState(m)
+                }
+
+                case ForgetDemultiplexingProcessingUnitMessage(id) => {
+                  // TODO This is a ugly hack of which I'm ashamed. /JD 20150428
```

Comment:
:-) I guess you could use e.g. become/unbecome to handle this a little better, or perhaps that's tricky with the AutoPilot..